gdx/src/com/badlogic/gdx/utils/Queue.java

```diff
@@ -142,8 +141,7 @@ public T removeFirst () {
 
 		final T result = values[head];
 		values[head] = null;
-		head++;
```

Comment:
So, I just want to jump in on this discussion.

First of all, I want to say that for readability, I prefer having the increment out of the if.

However, I mostly want to jump in because of the argument that this would create multiple reads. Fact is, you simply can't rely on what bytecode is generated for such small syntactic differences. I did test it out for the current java version on my system (javac 1.8.0_74), and when using an instance field it actually generated the exact same bytecode:

```
aload_0     //Load this pointer on stack
dup         //Duplicate this pointer on stack
getfield #2 // Field test:I
iconst_1    // Push 1 to stack
iadd        // Add top 2 elements
putfield #2 // Field test:I
aload_0     // Load this pointer on stack
getfield #2 // Field test:I
```
However, when using a static field, it would read the field only once, but duplicate it once more on the stack. It then depends on the jit compiler whether one is faster than the other. However, in the field instance case, performance will be exactly the same.